src/libexpr/include/nix/expr/eval-settings.hh

```diff
@@ -53,6 +53,13 @@ struct EvalSettings : Config
 
     std::vector<PrimOp> extraPrimOps;
 
+    Setting<bool> replaceEvalErrors{
```

Comment:
This should not be a setting but a flag to `nix eval`. We should avoid global settings that change what the output of commands looks like (e.g. `nix eval --json`), because then a user's settings might change the semantics of scripts that use those commands.